python/tvm/relay/frontend/onnx.py

```diff
@@ -351,6 +366,13 @@ class AveragePool(Pool):
     name = "avg_pool"
 
 
+class QLinearAveragePool(Pool):
```

Comment:
I think composition rather than subclassing would be a cleaner solution. Right now all the code to handle quantization + not quantization are in the same place which makes it a bit harder to read. Please separate it.

You can do something like refactor the Pool impl to a new class method like _run_calculation(...) and call it from QLinearAveragePool

Reply:
I've refactored the code to separate out the quantized and non-quantized ops in a way that doesn't touch any of the non-quantized op classes. Let me know if this is close to what you had in mind, thanks.

Reply:
Yeah this is fine